lib/cooked_post_processor.rb

```diff
@@ -32,7 +32,7 @@ def initialize(post, opts = {})
   def post_process(bypass_bump = false)
     DistributedMutex.synchronize("post_process_#{@post.id}") do
       DiscourseEvent.trigger(:before_post_process_cooked, @doc, @post)
-      removed_direct_reply_full_quotes
+      removed_direct_reply_full_quotes if SiteSetting.remove_full_quote
```

Comment:
The `SiteSetting.remove_full_quote` check should be done in the `removed_direct_reply_full_quotes` instead.